code/modules/fluids/fluid_objects.dm

```diff
@@ -467,11 +467,10 @@ TYPEINFO(/obj/item/sea_ladder)
 
 	afterattack(atom/target, mob/user as mob)
 		if (istype(target,/turf/space/fluid/warp_z5))
-			var/turf/space/fluid/warp_z5/hole = target
-			hole.try_build_turf_list() //in case we dont have one yet
-
-			deploy_ladder(hole, pick(hole.L), user)
-
+			SPAWN(0.1 SECONDS)	// give the turf's pitfall component time to work
```

Comment:
Why is this necessary? Work on what?

Reply:
there is a linking mechanism for holes and sea ladders. I moved it to the component seeing as it was related, but it also means that every time attackby is called, the `update_targets()` proc needs to run to update the list.

It's messy though

Reply:
Why not just call `update_targets()` here though?